packages/docusaurus-theme-classic/src/theme/DocPage/index.js

```diff
@@ -13,16 +13,26 @@ import renderRoutes from '@docusaurus/renderRoutes';
 import Layout from '@theme/Layout';
 import DocSidebar from '@theme/DocSidebar';
 import MDXComponents from '@theme/MDXComponents';
+import NotFound from '@theme/NotFound';
+import {matchPath} from '@docusaurus/router';
 
 import styles from './styles.module.css';
 
+function matchingRouteExist(routes, pathname) {
+  return routes.some(route => matchPath(pathname, route));
+}
+
 function DocPage(props) {
   const {route, docsMetadata, location} = props;
   const {permalinkToSidebar, docsSidebars} = docsMetadata;
   const sidebar = permalinkToSidebar[location.pathname.replace(/\/$/, '')];
   const {siteConfig: {themeConfig = {}} = {}} = useDocusaurusContext();
   const {sidebarCollapsible = true} = themeConfig;
 
+  if (!matchingRouteExist(route.routes, location.pathname)) {
```

Comment:
Wonder if there's a way to do this automatically for all plugins. Should it be done automatically?

Reply:
We shouldn't. Some theme might want different behavior. E.g: render a table of content of all available docs routes, or maybe display all available versions, etc. Very UI dependent